src/debugger/include/debugger/debugger_interface.h

```diff
@@ -0,0 +1,18 @@
+/**
+ * @file debugger_interface.h
+ * Declares the DebuggerCore Interface Class for CLI Operations
+ */
+
+#include "debugger/debugger.fwd.h"
```

Comment:
I thought of making IDebugger a friend class of DebuggerCore. But then I thought it's not necessary because we have public helper functions to retrieve private members if required.

Reply:
Ah okay.

An interface is a class you're never going to instantiate, so a friend class for it doesn't do much.

Reply:
Ah, okay got it.